docs/proposals/metadata.md

```diff
@@ -0,0 +1,153 @@
+# Image Metadata
+
+## Abstract
+
+A proposal for the additional metadata for Docker images which will provide more
+context about the content of the image and define relations between Docker
+images.
+
+## Motivation
+
+Image metadata are needed to do automatic generation of the OpenShift v3
+resource templates, like a PodTemplates, ReplicationControllers, BuildConfigs
+and other.  Automatic generation of PodTemplate is also required for
+Deployments.
+
+## Constraints and Assumptions
+
+For the time being, there is no good way how to define metadata for Docker
+images other than specifying extra environment variables. There is an upstream
+work currently being done to fix this situlation and we should switch to that in
+the implementation in the future.
+
+We also assume that there might be more use-cases and metadata define for images
+that helps with generation stuff, this document only provides those needed in
+the short-term.
```

Comment:
I would change this paragraph to "This document only defines the metadata needed by the current set of use cases. Additional metadata and/or use cases may be added in the future"